resources/js/components/UserRegister.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+
+// Assuming your CSS file is in the correct path
+import '@/css/registerpage.css';
+
+const Register = () => {
+    const [formData, setFormData] = useState({
+        username: '',
+        email: '',
+        password: '',
+        password_confirmation: '',
+    });
+
+    const [errors, setErrors] = useState({});
+    const [success, setSuccess] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+
+    const navigate = useNavigate(); // Hook for navigation
+
+    const onChange = e => {
+        setFormData({ ...formData, [e.target.name]: e.target.value });
+    };
+
+    const onSubmit = async e => {
+        e.preventDefault();
+        setErrors({}); // Reset errors on new submission
+        setSuccess(''); // Reset success message on new submission
+        setIsLoading(true);
+
+        try {
+            const response = await axios.post('http://localhost:8000/api/create_user', formData);
```

Comment:
localhost? I think we need to set variable and change it for the different environments

Reply:
Done